### ISSUE TITLE
🐛 Memory allocation of x bytes failed (macOS/Homebrew)

### ISSUE THREAD

**Issue by ZevEisenberg** [dandavison/delta#1201]
```
$ delta --help
memory allocation of 324400560136122233 bytes failed
[1]    80687 abort      delta --help
```

Exit code: 134

Xcode 13.4.1 (13F100) on macOS 12.6 (21G115)

This is on a new M1 Mac that I restored from a Time Machine backup of an old M1 Mac, so something could be messed up in my configuration, but I don't know what it would be.

This is similar to #1018, #1007, #878, but the fix for all of those involved package managers that I'm not using, so I'm not sure what the equivalent fix would be.

Here's the delta portion of my Git config:

```git-config
[core]
    trustctime = false
    pager = delta
    attributesfile = ~/.gitattributes
[interactive]
    diffFilter = delta --color-only
[delta]
    syntax-theme = Solarized (light)
    features = line-numbers decorations
    whitespace-error-style = 22 reverse
[delta "decorations"]
    commit-decorations-style = bold yellow box ul
    file-style = bold yellow ul
    file-decoration-style = none
```

**Comment by dandavison** [comment]
Hi @ZevEisenberg, the next step here would be to uninstall and then reinstall delta.

> This is on a new M1 Mac that I restored from a Time Machine backup of an old M1 Mac

I'm also not sure exactly what has gone wrong, but it does sound like the executable that was carried over is not happy.

**Comment by ZevEisenberg** [comment]
Oh, I forgot to mention. I did `brew uninstall` and then `brew install` it.

**Comment by ZevEisenberg** [comment]
Ooh, I should just uninstall and reinstall all of homebrew. It didn't occur to me that dependencies might need reinstallation as well. I'll also try the apps you suggested.

**Comment by ZevEisenberg** [comment]
Aha! Success!

Uninstalling and reinstalling homebrew didn't fix it. But installing bat revealed something interesting:

```
$ bat Brewfile
[bat error]: The binary caches for the user-customized syntaxes and themes in '/Users/igeek/.cache/bat' are not compatible with this version of bat (0.22.1). To solve this, either rebuild the cache (bat cache --build) or remove the custom syntaxes/themes (bat cache --clear).
For more information, see:

  https://github.com/sharkdp/bat#adding-new-syntaxes--language-definitions
```

That triggered some memory for me that maybe delta uses bat themes? I don't even know if it does, but anyway I did the `bat cache --build` and then both bat and delta started working again! I don't know if there's a way to detect this case in delta.

I'm attaching a copy of ~/.cache/bat from an old backup that I restored this machine from. I'm not sure if they're the exact offending files, or if they'd repro the issue, but maybe they'll be helpful.

[bat.zip](https://github.com/dandavison/delta/files/9648662/bat.zip)
